src/app/clusters/device-energy-management-server/device-energy-management-server.cpp

```diff
@@ -654,6 +746,96 @@ void Instance::HandleRequestConstraintBasedForecast(HandlerContext & ctx,
+        return;
+    }
+
+    // Check for invalid power levels and whether the constraint time/duration is in the past
+    {
+        auto iterator = constraints.begin();
+        if (iterator.Next())
+        {
+            const Structs::ConstraintsStruct::DecodableType & constraint = iterator.GetValue();
+
+            // Check to see if this constraint is in the past
+            if (constraint.startTime < currentUtcTime)
+            {
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+                return;
+            }
+
+            if (HasFeature(Feature::kPowerForecastReporting))
+            {
+                if (!constraint.nominalPower.HasValue())
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast no nominalPower");
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+                    return;
+                }
+
+                if (constraint.nominalPower.Value() < mDelegate.GetAbsMinPower() ||
+                    constraint.nominalPower.Value() > mDelegate.GetAbsMaxPower())
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast nominalPower out of range [absMinPower, absMaxPower]");
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+                    return;
+                }
+
+                if (!constraint.maximumEnergy.HasValue())
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast no value for maximumEnergy");
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+                    return;
+                }
+            }
+
+            if (HasFeature(Feature::kStateForecastReporting))
+            {
+                if (!constraint.loadControl.HasValue())
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast no loadControl");
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+                    return;
+                }
+
+                if (constraint.loadControl.Value() < -100 || constraint.loadControl.Value() > 100)
+                {
+                    ChipLogError(Zcl, "DEM: RequestConstraintBasedForecast bad loadControl %d", constraint.loadControl.HasValue());
+                    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+                    return;
+                }
+            }
+        }
+    }
+
+    // Check for overlappping elements
+    {
+        auto iterator = constraints.begin();
+        if (iterator.Next())
+        {
+            // Get the first constraint
+            const Structs::ConstraintsStruct::DecodableType & prevConstraint = iterator.GetValue();
+
+            // Start comparing next vs prev constraints
+            while (iterator.Next())
```

Comment:
Status check?

But also, after this call, `prevConstraint` has been overwritten.  So I would expect the tests below to always fail.  Was this tested?